sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -100,6 +101,17 @@ object TypeCoercion {
     case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
       Some(TimestampType)
 
+    case (t1 @ StructType(fields1), t2 @ StructType(fields2)) if t1.sameType(t2) =>
+      Some(StructType(fields1.zip(fields2).map { case (f1, f2) =>
+        // Since `t1.sameType(t2)` is true, two StructTypes have the same DataType
+        // except `name` (in case of `spark.sql.caseSensitive=false`) and `nullable`.
+        // - Different names: use a lower case name because findTightestCommonType is commutative.
+        // - Different nullabilities: `nullable` is true iff one of them is nullable.
+        val name = if (f1.name == f2.name) f1.name else f1.name.toLowerCase(Locale.ROOT)
```

Comment:
Why the output nested column name is lower case? Is Hive behaving like this?

In addition, could you add one more test and check whether we also respect case sensitivity conf when we resolve the queries that contain the nested column in the references?

Reply:
Yes. Hive does like the following.
```sql
hive> CREATE TABLE S AS SELECT named_struct('A',1);
hive> DESCRIBE S;
OK
_c0                 	struct<a:int>
```

Reply:
For a test case, does `nested column in the references` mean `WHERE` clause?

Reply:
Do both `tab.Col1.a` and `tab.Col1.A`work well when case sensitivity is off?

Reply:
When case sensitive is off, Spark considers them in lower cases.
For example, in [the test case](https://github.com/apache/spark/pull/18460/files#diff-5a2e7f03d14856c8769fd3ddea8742bdR2657), we need table name `struct1` and `struct2`. In case of `a = A`, it raises ambiquous column exceptions.
```
checkAnswer(sql("SELECT * FROM struct1, struct2 WHERE struct1.a = struct2.A"), Seq.empty)
```

Reply:
I mean, does the case sensitivity conf works in nest column name resolution?

Reply:
For Hive, your example works like the following.
```
hive> CREATE TABLE S1 AS SELECT named_struct('A',1) Col1;
hive> SELECT S1.Col1.a, S1.Col1.A FROM S1;
OK
1	1
```

Reply:
Oh, I see. I thought so. Let me check that again.
> I mean, does the case sensitivity conf works in nest column name resolution?

Reply:
Yes. It works. I updated the [test cases](https://github.com/apache/spark/pull/18460/files#diff-5a2e7f03d14856c8769fd3ddea8742bdR2657).